### PR TITLE
Tf lite gpu unity

### PR DESCRIPTION
**Pull request by stephen-systemfriend**
Changes to the unity Interpreter script to allow optional use of the GPU delegate on android for tensorflow, and corresponding updates to the readme to explain the additional build changes required.

### CODE REVIEW

**Comment by stephen-systemfriend** [comment]
@googlebot I signed it!

File: tensorflow/lite/experimental/examples/unity/TensorFlowLitePlugin/Assets/TensorFlowLite/SDK/Scripts/Interpreter.cs [hunk 1]
```diff
@@ -152,7 +173,67 @@ public class Interpreter : IDisposable
         TfLiteTensor tensor,
         IntPtr output_data,
         int output_data_size);
+    
+    enum TfLiteGpuInferenceUsage : Int32 {
```
Comment (jdduke) [inline tensorflow/lite/experimental/examples/unity/TensorFlowLitePlugin/Assets/TensorFlowLite/SDK/Scripts/Interpreter.cs]: Nice! Thanks for pushing forward on this, really cool that this "just works"!

Could we add the GpuDelegate in its own class/file (and make a Delegate interface in a separeate file)? We can pattern it after what we've done in Java, where Delegate.java has a simple interface to return the native handle to the native delegate instance. We can then pass in the Delegate class as an argument to the Interpreter constructor.

**Review by jdduke — CHANGES_REQUESTED** [review]

File: tensorflow/lite/experimental/examples/unity/TensorFlowLitePlugin/Assets/TensorFlowLite/SDK/Scripts/Interpreter.cs (hunk 1 above)
Comment (stephen-systemfriend) [reply tensorflow/lite/experimental/examples/unity/TensorFlowLitePlugin/Assets/TensorFlowLite/SDK/Scripts/Interpreter.cs]: Sure, I'll get on that and test it again

Comment (stephen-systemfriend) [reply tensorflow/lite/experimental/examples/unity/TensorFlowLitePlugin/Assets/TensorFlowLite/SDK/Scripts/Interpreter.cs]: The name delegate is already used in C#. Does TFDelegate work here?

Comment (jdduke) [reply tensorflow/lite/experimental/examples/unity/TensorFlowLitePlugin/Assets/TensorFlowLite/SDK/Scripts/Interpreter.cs]: Let's call it InterpreterDelegate for now (for the interface), and GpuDelegate (for the implementation).

**Comment by stephen-systemfriend** [comment]
hmm, seems like the delegate doesn't just work as is as well as I thought. On further inspection it seems the delegate failed to initialise properly and then it fell back to the cpu, which is why I thought it worked the first time. I'll continue investigating, but we should probably close this pull request as the scope may be larger

**Pull request closed without merging**